### DISCUSSION TITLE
change "Probable account sharing" to "Login from different IP addresses"

### DISCUSSION THREAD

**Discussion by sanderjo** [sabnzbd/sabnzbd#2354] [Ideas]
Triggered by https://www.reddit.com/r/usenet/comments/z7jm24/sabnzbd_gave_likely_password_sharing_for/

SAB can say "Probable account sharing"
I think we should change that to the more factually correct "Login from different IP addresses"
Reason: possible causes are not really account-sharing, but:

- you switch between VPN and no-VPN 
- your ISP uses "CGNAT". T-Mobile USA does that. And more ISPs will introduce CGNAT because of IPv4 prices
- your account is used from different places
- your download clients switches between IPv4 and IPv6. I've seen this with eweka. Not nice of eweka. EDIT: I cannot reproduce it with eweka (with newsreader and news6 at the same prio), so maybe I must revoke this possible cause.

Furthermore, I think the error message should have a link to sabnzbd.org explaining the possible causes mentioned above.

And looking at the code, I think the error message from the newsserver is now not included, but we should.

Now:
`Probable account sharing (news.eweka.nl)`

change to
`Login from different IP addresses (news.eweka.nl) <message from newsserver>. See https://sabnzbd.org/blabla`

sabnzbd/downloader.py has this code
```
                            elif error.code in (502, 481, 482) and clues_too_many_ip(error.msg):
                                # Account sharing?
                                if server.active:
                                    errormsg = T("Probable account sharing") + display_msg
                                    if server.errormsg != errormsg:
                                        server.errormsg = errormsg
                                        name = " (%s)" % server.host
                                        logging.warning(T("Probable account sharing") + name)
                                penalty = _PENALTY_SHARE
                                block = True
```
with
```
def clues_too_many_ip(text: str) -> bool:
    """Check for any "account sharing" clues in the response code"""
    text = text.lower()
    for clue in ("simultaneous ip", "multiple ip"):
        if clue in text:
            return True
    return False
```

**Comment by puzzledsab** [comment]
I agree, only issue is that it requires a new translation. A URL to an explanation should definitely be added.

**Reply by sanderjo** [reply]
> only issue is that it requires a new translation.

@Safihre acceptable for you?

**Comment by sanderjo** [comment]
I tried with eweka with logins from two different machines with a handmade script, and I get:

`nntplib.NNTPPermanentError: 502 (max number of simultaneous IP addresses reached: 2)`

Ok, 502 is clear, and eweka says something about IP addresses ("simultaneous IP", which is factual correct), which results in `True` from `clues_too_many_ip()` 
So including that in the SAB message is useful.

But ... "max number of simultaneous IP addresses reached: 2" ... based on that, I would expect that 2 is allowed. But apparantly 2 is too many. Anyway.

**Comment by sanderjo** [comment]
About URL and Wiki:

As reference: existing text/URL for incomplete:
![image](https://user-images.githubusercontent.com/1273502/206738113-3225d8e2-be7a-4c06-a622-c564f2b2f7f8.png)

https://sabnzbd.org/not-complete which then redirects to https://sabnzbd.org/wiki/introduction/downloads-cannot-be-completed.html

So, for "multiple IP addresses": how about:

"Multiple IP addresses" with URL https://sabnzbd.org/multiple-adresses with redirect to https://sabnzbd.org/wiki/introduction/login-from-multiple-ip-addresses.html

**Comment by sanderjo** [comment]
Prototype in SABnzbd:

![image](https://user-images.githubusercontent.com/1273502/206743777-e005aaf7-5c9e-47ee-b794-ce50821bb7d0.png)


![image](https://user-images.githubusercontent.com/1273502/206743738-10495f53-c294-4387-b5b2-aa2a341a2554.png)

**Comment by sanderjo** [comment]
First version of Wiki info:
short cut https://sabnzbd.org/multiple-adresses
leading to https://sabnzbd.org/wiki/introduction/login-from-multiple-ip-addresses.html

**Comment by sanderjo** [comment]
Worth noting: eweka does **not** block on a simultaneous login from IPv4 and IPv6:
`Serversnews.eweka.nl=852.3 MB, news6.eweka.nl=209.9 MB`

That's nice of eweka and good to know.

**Comment by sanderjo** [comment]
![image](https://user-images.githubusercontent.com/1273502/207117986-3f67ff0d-9efe-4fb8-9672-0420f9846da1.png)

For reference: the old one:

![image](https://user-images.githubusercontent.com/1273502/207118162-b3d3b241-6e7c-4f1d-9f5f-30544876364e.png)

**Comment by Safihre** [comment]
Does this really happen often enough that we need a wiki page for it?
If we do have a wiki page, I would at least expect a simple explanttion what this CGNAT is.. We don't all work at ISP's @sanderjo 😉

**Reply by sanderjo** [reply]
I suggested the URL / wiki on Dec 1 (13 days ago) and heard no objection then? New insight?

Explaining CGNAT on the SAB wiki is easy, but let's first answer the question about "often enough":

https://www.google.com/search?q=%22Probable+account+sharing%22 ... 136 hits
https://www.google.com/search?q=%22Probably+account+sharing%22 ... 1100 hits
https://www.google.com/search?q=+%22account+sharing%22+sabnzbd ... 1600 hits

Is that often enough?
